nftsentry/src/license.rs

```diff
@@ -99,13 +108,20 @@ impl Contract {
     }
 
     #[payable]
-    pub fn nft_propose_license(&mut self, authorized_id: Option<String>,token_id: TokenId, proposed_license: TokenLicense, receiver_id: AccountId){
+    pub fn nft_propose_license(&mut self, token_id: TokenId, proposed_license: TokenLicense){
        //measure the initial storage being used on the contract
         let initial_storage_usage = env::storage_usage();
         
-        let master_id = env::predecessor_account_id();
+        let predecessor_id = env::predecessor_account_id();
+        let current_account_id = env::current_account_id();
+        let token = self.tokens_by_id.get(&token_id).expect("No token");
+
+
+        if predecessor_id != current_account_id {
+            panic!("Only the contract owner can propose a license update");
```

Comment:
current account ID is not always a contract owner, e.g. acc1 can be an owner of contract1 while acc1 != contract1. For this purpose `self.owner_id` needs to be used

Reply:
I'm removing the checks per the slack conversation.